File: examples/modulith-clinical-tool-jpa/src/test/java/io/zenwave360/example/clinicaltool/modules/surveys/repository/jpa/QuestionRepositoryIntegrationTest.java
package io.zenwave360.example.clinicaltool.modules.surveys.repository.jpa;

import io.zenwave360.example.clinicaltool.common.BaseRepositoryIntegrationTest;
import io.zenwave360.example.clinicaltool.modules.surveys.domain.*;
import jakarta.persistence.EntityManager;
import java.time.*;
import java.util.List;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;

class QuestionRepositoryIntegrationTest extends BaseRepositoryIntegrationTest {

    @Autowired
    EntityManager entityManager;

    @Autowired
    QuestionRepository questionRepository;

    @Test
    void findAllTest() {
        var results = questionRepository.findAll();
        Assertions.assertFalse(results.isEmpty());
    }

    @Test
    void findByIdTest() {
        var id = 1L;
        var question = questionRepository.findById(id).orElseThrow();
        Assertions.assertNotNull(question.getId());
        Assertions.assertNotNull(question.getVersion());
        Assertions.assertNotNull(question.getCreatedBy());
        Assertions.assertNotNull(question.getCreatedDate());
    }

    @Test
    void saveTest() {
        Question question = new Question();
        question.setName("");
        question.setQuestionType(QuestionType.values()[0]);
        question.setRequired(true);
        question.setRangeStart(0);
        question.setRangeEnd(0);
        question.setTranslations(List.of(new QuestionTranslation()));
        question.setOptions(List.of(new Option()));
        question.setIncludeOther(false);

        // Persist aggregate root
        var created = questionRepository.save(question);

        // reloading to get relationships persisted by id
        entityManager.flush();
        entityManager.refresh(created);
        Assertions.assertNotNull(created.getId());
        Assertions.assertNotNull(created.getVersion());
        Assertions.assertNotNull(created.getCreatedBy());
        Assertions.assertNotNull(created.getCreatedDate());
    }

    @Test
    void updateTest() {
        var id = 1L;
        var question = questionRepository.findById(id).orElseThrow();
        question.setName("");
        question.setQuestionType(QuestionType.values()[0]);
        question.setRequired(true);
        question.setRangeStart(0);
        question.setRangeEnd(0);
        question.setTranslations(List.of(new QuestionTranslation()));
        question.setOptions(List.of(new Option()));
        question.setIncludeOther(false);

        question = questionRepository.save(question);
        Assertions.assertEquals("", question.getName());
        Assertions.assertEquals(QuestionType.values()[0], question.getQuestionType());
        Assertions.assertEquals(true, question.isRequired());
        Assertions.assertEquals(0, question.getRangeStart());
        Assertions.assertEquals(0, question.getRangeEnd());
        Assertions.assertEquals(List.of(new QuestionTranslation()), question.getTranslations());
        Assertions.assertEquals(List.of(new Option()), question.getOptions());
        Assertions.assertEquals(false, question.getIncludeOther());
    }

    @Test
    void deleteTest() {
        var id = 1L;
        questionRepository.deleteById(id);
        var notFound = questionRepository.findById(id);
        Assertions.assertFalse(notFound.isPresent());
    }
}

File: examples/modulith-klinical-tool-jpa/src/test/kotlin/io/example/asyncapi/provider/clinicaltool/modules/surveys/repository/jpa/QuestionRepositoryIntegrationTest.kt
package io.example.asyncapi.provider.clinicaltool.modules.surveys.repository.jpa

import io.zenwave360.example.clinicaltool.common.BaseRepositoryIntegrationTest
import io.zenwave360.example.clinicaltool.modules.surveys.domain.*
import io.zenwave360.example.clinicaltool.modules.surveys.repository.jpa.QuestionRepository

import java.util.HashSet
import java.util.HashMap
import java.util.List
import java.time.*
import java.math.BigDecimal

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired

import jakarta.persistence.EntityManager

class QuestionRepositoryIntegrationTest : BaseRepositoryIntegrationTest() {

    @Autowired
    lateinit var entityManager: EntityManager

    @Autowired
    lateinit var questionRepository: QuestionRepository

    @Test
    fun findAllTest() {
        val results = questionRepository.findAll()
        Assertions.assertFalse(results.isEmpty())
    }

    @Test
    fun findByIdTest() {
        val id = 1L
        val question = questionRepository.findById(id).orElseThrow()
        Assertions.assertNotNull(question.id)
        Assertions.assertNotNull(question.version)
        Assertions.assertNotNull(question.createdBy)
        Assertions.assertNotNull(question.createdDate)
    }

    @Test
    fun saveTest() {
        val question = Question()
        question.name = ""
        question.questionType = QuestionType.values()[0]
        question.required = true
//        question.rangeStart = java.lang.Integer.valueOf(42)
//        question.rangeEnd = 0.toInt()
        question.translations = List.of(QuestionTranslation())
        question.options = List.of(Option())
        question.includeOther = false



        // Persist aggregate root
        val created = questionRepository.save(question)

        // reloading to get relationships persisted by id
        entityManager.flush()
        entityManager.refresh(created)
        Assertions.assertNotNull(created.id)
        Assertions.assertNotNull(created.version)
        Assertions.assertNotNull(created.createdBy)
        Assertions.assertNotNull(created.createdDate)


    }

    @Test
    fun updateTest() {
        val id = 1L
        val question = questionRepository.findById(id).orElseThrow()
        question.name = ""
        question.questionType = QuestionType.values()[0]
        question.required = true
//        question.rangeStart = 0.toInt()
//        question.rangeEnd = 0.toInt()
        question.translations = List.of(QuestionTranslation())
        question.options = List.of(Option())
        question.includeOther = false

        val updated = questionRepository.save(question)
        Assertions.assertEquals("", updated.name)
        Assertions.assertEquals(QuestionType.values()[0], updated.questionType)
        Assertions.assertEquals(true, updated.required)
        Assertions.assertEquals(0, updated.rangeStart)
        Assertions.assertEquals(0, updated.rangeEnd)
        Assertions.assertEquals(List.of(QuestionTranslation()), updated.translations)
        Assertions.assertEquals(List.of(Option()), updated.options)
        Assertions.assertEquals(false, updated.includeOther)
    }

    @Test
    fun deleteTest() {
        val id = 1L
        questionRepository.deleteById(id)
        val notFound = questionRepository.findById(id)
        Assertions.assertFalse(notFound.isPresent)
    }
}
